Aggregator/Aggregator.cs

```diff
@@ -0,0 +1,336 @@
+
+    }
+    /// <summary>
+    /// An instance of this class is created for each service replica by the Service Fabric runtime.
+    /// </summary>
+    internal sealed class Aggregator : StatefulService,IMyCommunication
+    {
+        protected CancellationToken token { get; set; }
+
+        public Aggregator(StatefulServiceContext context)
+            : base(context)
+        { }
+/// <summary>
+        /// This is the main entry point for your service replica.
+        /// This method executes when this replica of your service becomes primary and has write status.
+        /// </summary>
+        /// <param name="cancellationToken">Canceled when Service Fabric needs to shut down this service replica.</param>
+        protected override async Task RunAsync(CancellationToken cancellationToken)
+        {
+   
+            this.token = cancellationToken;
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                
+                await Task.Delay(TimeSpan.FromMilliseconds(SFUtilities.intervalMiliseconds), cancellationToken);
+                while(await GetMinQueueCountAsync() > 1)
+                {
+                    await ProduceSnapshotAsync();
+                    await TryPruneAsync(); 
+                }
+            }
+        }
+        public async Task PutDataRemote(string queueName,byte[] data)
+        {
+            await AddDataAsync(queueName, data);
+        }
+
+        public async Task<List<byte[]>> GetDataRemote(string queueName)
+        {
+            return await GetDataAsync(queueName);
+        }
+
+        
+        public async Task<List<Snapshot>> GetSnapshotsRemote(double milisecondsLow, double milisecondsHigh)
+        {
+            List<Snapshot> list = new List<Snapshot>();
+
+            var stateManager = this.StateManager;
+            var reliableQueue = await stateManager.GetOrAddAsync<IReliableQueue<byte[]>>(Snapshot.queueName);
+            try
+            {
+
+                using (var tx = stateManager.CreateTransaction())
+                {
+                    var iterator = (await reliableQueue.CreateEnumerableAsync(tx)).GetAsyncEnumerator();
+
+                    byte[] data = null;
+                    while (await iterator.MoveNextAsync(token))
+                    {
+                        data = iterator.Current;
+
+                        if (data != null) {
+                            Snapshot s = (Snapshot)ByteSerialization.ByteArrayToObject(data);
+                            if (s.miliseconds >= milisecondsLow && s.miliseconds <= milisecondsHigh) list.Add(s);
+                            else 
+                            {//help garbage collector to free memory
+                                s = null;
+                                data = null;
+                            }
+                        } 
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                var x = e;
```

Comment:
:-) (debugging...)

Reply:
Yes this catch can be removed the catch is never triggered